mrmustard/math/parameters.py

```diff
@@ -0,0 +1,275 @@
+    def __init__(self, value: any, name: str):
+        if math.from_backend(value) and not math.is_trainable(value):
+            self._value = value
+        elif type(value) in [list, int, float]:
+            self._value = math.new_constant(value, name)
+        else:
+            self._value = math.new_constant(value, name, value.dtype)
+
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        r"""
+        The name of this constant.
+        """
+        return self._name
+
+    @property
+    def value(self) -> any:
+        r"""
+        The value of this constant.
+        """
+        return self._value
+
+    def __mul__(self, value):
+        return type(self)(value=value * self.value, name=self.name)
+
+    def __rmul__(self, value):
+        return type(self)(value=self.value * value, name=self.name)
+
+
+# pylint: disable=too-many-instance-attributes
+class Variable:
+    r"""
+    A parameter whose value can change.
+
+    .. code::
+
+      my_var = Variable(1, "my_var")
+
+    Args:
+        value: The value of this variable.
+        name: The name of this variable.
+        bounds: The numerical bounds of this variable.
+        update_fn: The function used to update this variable during training.
+    """
+
+    def __init__(
+        self,
+        value: any,
+        name: str,
+        bounds: Tuple[Optional[float], Optional[float]] = (None, None),
+        update_fn: Callable = update_euclidean,
+    ):
+        self._value = self._get_value(value, bounds, name)
+        self._name = name
+        self._bounds = bounds
+        self._update_fn = update_fn
+
+    def _get_value(self, value, bounds, name):
+        r"""
+        Returns a variable from given ``value``, ``bounds``, and ``name``.
+        """
+        if math.from_backend(value) and math.is_trainable(value):
+            return value
+        elif type(value) in [list, int, float]:
+            return math.new_variable(value, bounds, name)
+        else:
+            return math.new_variable(value, bounds, name, value.dtype)
+
+    @property
+    def bounds(self) -> Tuple[Optional[float], Optional[float]]:
+        r"""
+        The numerical bounds of this variable.
+        """
+        return self._bounds
+
+    @property
+    def name(self) -> str:
+        r"""
+        The name of this constant.
+        """
+        return self._name
+
+    @property
+    def update_fn(self) -> Optional[Callable]:
+        r"""
+        The function used to update this variable during training.
+        """
+        return self._update_fn
+
+    @update_fn.setter
+    def update_fn(self, value):
+        self._update_fn = value
+
+    @property
+    def value(self) -> any:
+        r"""
+        The value of this variable.
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        self._value = self._get_value(value, self.bounds, self.name)
+
+    @staticmethod
+    def orthogonal(
+        value: any,
```

Comment:
if value=None, can we initialize a random one? We have the code already in math. Same for unitary and symplectic